### PR TITLE
Gallery prefab

### PR DESCRIPTION
**Pull request by ct-martin**
Some code cleanup/organization. D.R.Y. (don't repeat yourself) for image handling.

This will need to be documented, but images should now be included in the YAML front matter & use `{% include content-blocks/gallery.html %}` to show. Doing this creates a fancy-ish gallery & integrates with SEO. This is best used on project and event pages, but not so much for blog posts. Additionally, things like build status images (like on the Teleirc) page should not be use with this. If a gallery include is not in a page, but the page still has the images section of the front matter, it will integrate with SEO. The block has logic for if there is only one image (so still only need one reference to the image and needs no controls, but can still SEO) as well as a safeguard for if it's accidentally included on a page with no images set (in which case it will simply do nothing).

As two examples of this, the sample event page of Tim's Flock talk & the TigerOS page have been converted to use this block. Tim's page also demonstrates that when there is only one image it effectively acts as a responsive image embed.

### CODE REVIEW

**Comment by ct-martin** [comment]
Another point to mention, the front matter is also used on index pages (such as `https://ritlug.com/events`) like with SEO (uses first image).

**Comment by ct-martin** [comment]
And as a (hopefully final) note, the index & SEO functionality already exists on the site, this completes tying the whole system together in a cleaner fashion.

**Comment by Tjzabel** [comment]
I would highly suggest adding a comment of bullet points on the [website runbook issue](https://github.com/RITlug/runbook/issues/52) where you can give an overview of what needs to be documented on the website so you don't forget what you need to actually document.

**Comment by Tjzabel** [comment]
As far as a review, everything looks good to me. I know *absolutely nothing* about website development, so I will refer to @jwflory for a review on this (I think he knows a little web dev).

**Comment by ct-martin** [comment]
@Tjzabel: @jwflory isn't too familiar with the Jekyll aspects of developing the site (yet; I need to write some docs next month). On the contrary, @jrtechs has started learning how Jekyll works.

File: projects/_posts/2018-01-02-tigeros.html [hunk 1]
```diff
@@ -30,37 +30,7 @@ <h2>Download</h2>
...
-            <a class="carousel-control-prev" href="#carouselIndicators" role="button" data-slide="prev">
-                <span class="carousel-control-prev-icon" aria-hidden="true"></span>
-                <span class="sr-only">Previous</span>
-            </a>
-            <a class="carousel-control-next" href="#carouselIndicators" role="button" data-slide="next">
-                <span class="carousel-control-next-icon" aria-hidden="true"></span>
-                <span class="sr-only">Next</span>
-            </a>
-        </div>
+    <section id="screenshots">
```
Comment (jrtechs) [inline projects/_posts/2018-01-02-tigeros.html]: Very clean way of doing images.

File: events/_posts/2018-08-08-tim-flock.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ images:
 ---
 RITlug President Tim Zabel presents on TigerOS & Remixes at Fedora Flock 2018 in Dresden.
 
-<img src="/events/assets/2018/08/tjzabel-flock.jpg" class="w-100" style="margin-bottom:1.2em;" />
+{% include content-blocks/gallery.html %}
```
Comment (jrtechs) [inline events/_posts/2018-08-08-tim-flock.md]: Nice!

File: _includes/content-blocks/gallery.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{% if page.images %}
```
Comment (jrtechs) [inline _includes/content-blocks/gallery.html]: This will definitely help us eliminate "copy pasta" when creating new posts.

**Comment by ct-martin** [comment]
Merging since @jrtechs approved and I want to get some more done tonight

**Pull request merged**